### ISSUE TITLE
SwaggerUI throws an error when changing Servers

### ISSUE THREAD

**Issue by ponelat** [swagger-api/swagger-ui#7525]
SwaggerUI throws error: `Uncaught TypeError: Failed to construct 'URL': Invalid base URL`
When changing server (from the dropdown)

Steps:
- Load the following API definition: https://gist.github.com/ponelat/71dba4b85b403455e0718383f7cf3bd3
- Change the server from dropdown
Expected:
The server field is changed
Actual:
An Uncaught TypeError is thrown

SwaggerUI demo with definition: https://petstore.swagger.io/?url=https://gist.githubusercontent.com/ponelat/71dba4b85b403455e0718383f7cf3bd3/raw/50737b50e2ee42ebfc5afa7fb5137b588c6ba0cc/server-url-issue.yml#/default/get_something

**Comment by char0n** [comment]
Right, I can confirm on v4 as well. It crashes the SwaggerUI entirely as there is no error boundary to catch this error. I'm incorporating it for v4 bugfixes.

**Comment by mathis-m** [comment]
@char0n @ponelat had a quick check on this. Looks like this is the expected behavior, because the spec is invalid!
As [OAS](https://spec.openapis.org/oas/v3.1.0#:~:text=the%20contact%20information.-,This%20MUST%20be%20in%20the%20form%20of%20a%20URL,-.) states the info.contact.url MUST be in form of an URL. 
So just change `url: contact` to `url: /contact`:
```yaml
openapi: 3.0.0
info:
  title: Test API
  version: 1.0.0
  contact:
    name: Contact us
    url: /contact

servers:
  - url: https://api.example.com/foo/bar
  - url: https://api.example.com:{port}/{basePath}
    variables:
      port:
        default: '4443'
      basePath:
        default: v1

paths:
  /something:
    get:
      responses:
        '200':
          description: ok
```

**Comment by char0n** [comment]
@mathis-m thanks for having look at it!

Given your findings, it's reasonable to assume that not having a valid URL in `info.contact.url` will not crash the SwaggerUI while changing server URL.

**Comment by char0n** [comment]
@hkosova is right.

The original error is: `Uncaught TypeError: Failed to construct 'URL': Invalid base URL`

```js
new URL("contact", "https://api.example.com:{port}/{basePath}")
```

This  code throws exactly that error. Possible problem could be that the Server URL is used for resolution before it's variables gets trancluded. But we have to dive deep into code to see what actually causes this error.

**Comment by mathis-m** [comment]
@char0n your linkedin post reminded me of this issue. I had a quick check and it is obvious that if port is in template syntax it is not possible to parse it in the right way.

**Comment by char0n** [comment]
PR with bugfix has been issued. Cause of the problem and it's solution have been provided in PR description
